source/fab/parse/fortran.py

```diff
@@ -284,15 +285,38 @@ def _process_comment(self, analysed_file, obj):
         # TODO: error handling in case we catch a genuine comment
         # TODO: separate this project-specific code from the generic f analyser?
         depends_str = "DEPENDS ON:"
-        if depends_str in obj.items[0]:
+        comment = obj.items[0].strip()
+        if depends_str in comment:
             self.depends_on_comment_found = True
-            dep = obj.items[0].split(depends_str)[-1].strip()
+            dep = comment.split(depends_str)[-1].strip()
             # with .o means a c file
             if dep.endswith(".o"):
                 analysed_file.mo_commented_file_deps.add(dep.replace(".o", ".c"))
             # without .o means a fortran symbol
             else:
                 analysed_file.add_symbol_dep(dep)
+        if comment[:2] == "!$":
+            # Check if it is a use statement with an OpenMP sentinel:
+            # Use fparser's string reader to discard potential comment
+            # TODO #13: once fparser supports reading the sentinels,
+            # this can be removed.
```

Comment:
Will there be a problem if this code is used with an fparser with the directive support implemented?

Reply:
No. If we enable the handling of omp sentinels, no comment will be returned from fparser (instead the proper Fortran tree will be used).

The only inconsistent behaviour would be if we do not enable sentinels in fparser, but leave this code in I'd guess. Note that my intention is that the config object will be aware if the build is with or without openmp (since omp flags are compiler-specific, so this way the config object can trigger adding the appropriate flag in the compiler tools - either enable or disable openmp, e.g. Cray has openmp enabled by default).